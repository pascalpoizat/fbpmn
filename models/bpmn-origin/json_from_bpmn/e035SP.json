{
    "isInterrupt": {},
    "catN": {
        "EndEvent_0a59w28": "NoneEndEvent",
        "EndEvent_1jzyo34": "NoneEndEvent",
        "StartEvent_1": "NoneStartEvent",
        "StartEvent_0943q6x": "NoneStartEvent",
        "StartEvent_09ojxru": "MessageStartEvent",
        "P_": "Process",
        "EndEvent_0atuxkh": "NoneEndEvent",
        "SubProcess_1glz8ii": "SubProcess",
        "Q_": "Process",
        "Task_1x5zvyz": "AbstractTask",
        "Task_0mft1gb": "SendTask"
    },
    "catE": {
        "SequenceFlow_0pgod6g": "NormalSequenceFlow",
        "SequenceFlow_12utbfs": "NormalSequenceFlow",
        "SequenceFlow_00j4ong": "NormalSequenceFlow",
        "MessageFlow_183e2wp": "MessageFlow",
        "SequenceFlow_0yhgli1": "NormalSequenceFlow",
        "SequenceFlow_11rsxtg": "NormalSequenceFlow",
        "SequenceFlow_1uo79pd": "NormalSequenceFlow"
    },
    "containN": {
        "P_": [
            "StartEvent_1",
            "SubProcess_1glz8ii",
            "EndEvent_0a59w28"
        ],
        "SubProcess_1glz8ii": [
            "Task_1x5zvyz",
            "EndEvent_0atuxkh",
            "StartEvent_09ojxru"
        ],
        "Q_": [
            "StartEvent_0943q6x",
            "EndEvent_1jzyo34",
            "Task_0mft1gb"
        ]
    },
    "messageE": {
        "MessageFlow_183e2wp": "m1"
    },
    "timeInformation": {},
    "targetE": {
        "SequenceFlow_0pgod6g": "EndEvent_1jzyo34",
        "SequenceFlow_12utbfs": "SubProcess_1glz8ii",
        "SequenceFlow_00j4ong": "EndEvent_0a59w28",
        "MessageFlow_183e2wp": "StartEvent_09ojxru",
        "SequenceFlow_0yhgli1": "Task_1x5zvyz",
        "SequenceFlow_11rsxtg": "Task_0mft1gb",
        "SequenceFlow_1uo79pd": "EndEvent_0atuxkh"
    },
    "name": "e035SP",
    "attached": {},
    "containE": {
        "P_": [
            "SequenceFlow_12utbfs",
            "SequenceFlow_00j4ong"
        ],
        "SubProcess_1glz8ii": [
            "SequenceFlow_1uo79pd",
            "SequenceFlow_0yhgli1"
        ],
        "Q_": [
            "SequenceFlow_11rsxtg",
            "SequenceFlow_0pgod6g"
        ]
    },
    "edges": [
        "MessageFlow_183e2wp",
        "SequenceFlow_12utbfs",
        "SequenceFlow_00j4ong",
        "SequenceFlow_1uo79pd",
        "SequenceFlow_0yhgli1",
        "SequenceFlow_11rsxtg",
        "SequenceFlow_0pgod6g"
    ],
    "messages": [
        "m1"
    ],
    "sourceE": {
        "SequenceFlow_0pgod6g": "Task_0mft1gb",
        "SequenceFlow_12utbfs": "StartEvent_1",
        "SequenceFlow_00j4ong": "SubProcess_1glz8ii",
        "MessageFlow_183e2wp": "Task_0mft1gb",
        "SequenceFlow_0yhgli1": "StartEvent_09ojxru",
        "SequenceFlow_11rsxtg": "StartEvent_0943q6x",
        "SequenceFlow_1uo79pd": "Task_1x5zvyz"
    },
    "nodes": [
        "P_",
        "Q_",
        "StartEvent_1",
        "SubProcess_1glz8ii",
        "EndEvent_0a59w28",
        "Task_1x5zvyz",
        "EndEvent_0atuxkh",
        "StartEvent_09ojxru",
        "StartEvent_0943q6x",
        "EndEvent_1jzyo34",
        "Task_0mft1gb"
    ],
    "nameN": {
        "SubProcess_1glz8ii": "SP"
    }
}