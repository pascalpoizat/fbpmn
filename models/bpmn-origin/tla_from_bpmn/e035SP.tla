---------------- MODULE e035SP ----------------

EXTENDS TLC, PWSTypes

VARIABLES nodemarks, edgemarks, net

Interest ==
  "P_" :> { "m1" }
  @@ "Q_" :> {  }

ContainRel ==
  "P_" :> { "StartEvent_1", "SubProcess_1glz8ii", "EndEvent_0a59w28" }
  @@ "Q_" :> { "StartEvent_0943q6x", "EndEvent_1jzyo34", "Task_0mft1gb" }
  @@ "SubProcess_1glz8ii" :> { "Task_1x5zvyz", "EndEvent_0atuxkh", "StartEvent_09ojxru" }

Node == { "P_", "Q_", "StartEvent_1", "SubProcess_1glz8ii", "EndEvent_0a59w28", "Task_1x5zvyz", "EndEvent_0atuxkh", "StartEvent_09ojxru", "StartEvent_0943q6x", "EndEvent_1jzyo34", "Task_0mft1gb" }

Edge == { "MessageFlow_183e2wp", "SequenceFlow_12utbfs", "SequenceFlow_00j4ong", "SequenceFlow_1uo79pd", "SequenceFlow_0yhgli1", "SequenceFlow_11rsxtg", "SequenceFlow_0pgod6g" }

Message == { "m1" }

msgtype ==
   "MessageFlow_183e2wp" :> "m1"


source ==
   "MessageFlow_183e2wp" :> "Task_0mft1gb"
@@ "SequenceFlow_12utbfs" :> "StartEvent_1"
@@ "SequenceFlow_00j4ong" :> "SubProcess_1glz8ii"
@@ "SequenceFlow_1uo79pd" :> "Task_1x5zvyz"
@@ "SequenceFlow_0yhgli1" :> "StartEvent_09ojxru"
@@ "SequenceFlow_11rsxtg" :> "StartEvent_0943q6x"
@@ "SequenceFlow_0pgod6g" :> "Task_0mft1gb"

target ==
   "MessageFlow_183e2wp" :> "StartEvent_09ojxru"
@@ "SequenceFlow_12utbfs" :> "SubProcess_1glz8ii"
@@ "SequenceFlow_00j4ong" :> "EndEvent_0a59w28"
@@ "SequenceFlow_1uo79pd" :> "EndEvent_0atuxkh"
@@ "SequenceFlow_0yhgli1" :> "Task_1x5zvyz"
@@ "SequenceFlow_11rsxtg" :> "Task_0mft1gb"
@@ "SequenceFlow_0pgod6g" :> "EndEvent_1jzyo34"

CatN ==
   "P_" :> Process
@@ "Q_" :> Process
@@ "StartEvent_1" :> NoneStartEvent
@@ "SubProcess_1glz8ii" :> SubProcess
@@ "EndEvent_0a59w28" :> NoneEndEvent
@@ "Task_1x5zvyz" :> AbstractTask
@@ "EndEvent_0atuxkh" :> NoneEndEvent
@@ "StartEvent_09ojxru" :> MessageStartEvent
@@ "StartEvent_0943q6x" :> NoneStartEvent
@@ "EndEvent_1jzyo34" :> NoneEndEvent
@@ "Task_0mft1gb" :> SendTask

CatE ==
   "MessageFlow_183e2wp" :> MessageFlow
@@ "SequenceFlow_12utbfs" :> NormalSeqFlow
@@ "SequenceFlow_00j4ong" :> NormalSeqFlow
@@ "SequenceFlow_1uo79pd" :> NormalSeqFlow
@@ "SequenceFlow_0yhgli1" :> NormalSeqFlow
@@ "SequenceFlow_11rsxtg" :> NormalSeqFlow
@@ "SequenceFlow_0pgod6g" :> NormalSeqFlow

LOCAL preEdges ==
  [ i \in {} |-> {}]
PreEdges(n,e) == preEdges[n,e]

PreNodes(n,e) == { target[ee] : ee \in preEdges[n,e] }
          \union { nn \in { source[ee] : ee \in preEdges[n,e] } : CatN[nn] \in { NoneStartEvent, MessageStartEvent } }

BoundaryEvent ==
  [ i \in {} |-> {}]

WF == INSTANCE PWSWellFormed
ASSUME WF!WellTyped
ASSUME WF!WellFormedness

ConstraintNode == TRUE \* none
ConstraintEdge == TRUE \* none
Constraint == TRUE     \* none
INSTANCE PWSConstraints
INSTANCE UserProperties
INSTANCE PWSSemantics

================================================================

