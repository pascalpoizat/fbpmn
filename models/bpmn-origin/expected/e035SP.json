{
  "e035SP": {
    "Network02FifoPair": {
      "Prop02Safe": {
        "value": true,
        "states": "15",
        "trans": "23",
        "depth": "7"
      },
      "Prop04MsgSound": {
        "value": false,
        "states": "15",
        "trans": "23",
        "depth": null
      },
      "Prop03Sound": {
        "value": false,
        "states": "15",
        "trans": "23",
        "depth": null
      },
      "Prop01Type": {
        "value": true,
        "states": "15",
        "trans": "23",
        "depth": "7"
      }
    },
    "Network03Causal": {
      "Prop02Safe": {
        "value": true,
        "states": "15",
        "trans": "23",
        "depth": "7"
      },
      "Prop04MsgSound": {
        "value": false,
        "states": "15",
        "trans": "23",
        "depth": null
      },
      "Prop03Sound": {
        "value": false,
        "states": "15",
        "trans": "23",
        "depth": null
      },
      "Prop01Type": {
        "value": true,
        "states": "15",
        "trans": "23",
        "depth": "7"
      }
    },
    "Network07RSC": {
      "Prop02Safe": {
        "value": true,
        "states": "15",
        "trans": "23",
        "depth": "7"
      },
      "Prop04MsgSound": {
        "value": false,
        "states": "15",
        "trans": "23",
        "depth": null
      },
      "Prop03Sound": {
        "value": false,
        "states": "15",
        "trans": "23",
        "depth": null
      },
      "Prop01Type": {
        "value": true,
        "states": "15",
        "trans": "23",
        "depth": "7"
      }
    },
    "Network05Outbox": {
      "Prop02Safe": {
        "value": true,
        "states": "15",
        "trans": "23",
        "depth": "7"
      },
      "Prop04MsgSound": {
        "value": false,
        "states": "15",
        "trans": "23",
        "depth": null
      },
      "Prop03Sound": {
        "value": false,
        "states": "15",
        "trans": "23",
        "depth": null
      },
      "Prop01Type": {
        "value": true,
        "states": "15",
        "trans": "23",
        "depth": "7"
      }
    },
    "Network04Inbox": {
      "Prop02Safe": {
        "value": true,
        "states": "15",
        "trans": "23",
        "depth": "7"
      },
      "Prop04MsgSound": {
        "value": false,
        "states": "15",
        "trans": "23",
        "depth": null
      },
      "Prop03Sound": {
        "value": false,
        "states": "15",
        "trans": "23",
        "depth": null
      },
      "Prop01Type": {
        "value": true,
        "states": "15",
        "trans": "23",
        "depth": "7"
      }
    },
    "Network01Bag": {
      "Prop02Safe": {
        "value": true,
        "states": "15",
        "trans": "23",
        "depth": "7"
      },
      "Prop04MsgSound": {
        "value": false,
        "states": "15",
        "trans": "23",
        "depth": null
      },
      "Prop03Sound": {
        "value": false,
        "states": "15",
        "trans": "23",
        "depth": null
      },
      "Prop01Type": {
        "value": true,
        "states": "15",
        "trans": "23",
        "depth": "7"
      }
    },
    "Network06Fifo": {
      "Prop02Safe": {
        "value": true,
        "states": "15",
        "trans": "23",
        "depth": "7"
      },
      "Prop04MsgSound": {
        "value": false,
        "states": "15",
        "trans": "23",
        "depth": null
      },
      "Prop03Sound": {
        "value": false,
        "states": "15",
        "trans": "23",
        "depth": null
      },
      "Prop01Type": {
        "value": true,
        "states": "15",
        "trans": "23",
        "depth": "7"
      }
    }
  }
}
